libs/android/build.gradle

```diff
@@ -0,0 +1,13 @@
+String getVersion() {
+  def compileSdkVersion = Integer.parseInt(project.android.compileSdkVersion.split("-")[1])
```

Comment:
What is the value returned by the `project.android.compileSdkVersion` property? I'm trying to understand why are we using the split here. Isn't the compileSdkVersion an integer by default?

Reply:
I had the exact same question and it made a lot of confusion to me. Here's what's happening, there are basically two ways to set the compileSdkVersion value through gradle: using an int and using a string. Basically, whenever you set using an int, [it goes into a setter that behind the curtain basically prepends "android-" ](https://android.googlesource.com/platform/tools/build/+/6d7fd0d2eff092abf1aaf44d03756b24570b390c/gradle/src/main/groovy/com/android/build/gradle/BaseExtension.groovy#112). 

If you set using a string, [it directly sets the value into the backing field](https://android.googlesource.com/platform/tools/build/+/6d7fd0d2eff092abf1aaf44d03756b24570b390c/gradle/src/main/groovy/com/android/build/gradle/BaseExtension.groovy#121).

Reply:
Thanks for the clarification! 
And you blame the Apple for their "stuff" 😑

Reply:
i think we can resolve this conversation so that noone sees what @luissilvaos is saying...